integration/run_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/awslabs/soci-snapshotter/util/testutil"
+)
+
+// TestRunMultipleContainersWithSameImage runs two containers with the same image at the same time
+func TestRunMultipleContainersWithSameImage(t *testing.T) {
+	RunMultipleWebContainers(t, []string{nginxImage, nginxImage})
+}
+
+// TestRunMultipleContainersWithDifferentImages runs two containers with different images at the same time
+func TestRunMultipleContainersWithDifferentImages(t *testing.T) {
+	RunMultipleWebContainers(t, []string{nginxImage, drupalImage})
+}
```

Comment:
We should use table driven tests here. You can take a look at how we've implemented other integration tests.

Reply:
Similar to https://github.com/awslabs/soci-snapshotter/blob/main/integration/pull_test.go#L91-L114 ?

Reply:
Do we have any table driven tests that do things in parallel?

Reply:
No, I don't think so.

Reply:
Are you suggesting that the two tests I've implemented could be a single test with two sub-tests, each driven from a table, or that each of my tests should have its list of containers to run as a table? If the latter, I can't see how to map the existing examples for table driven subtests to this situation where the containers need to run in parallel and make up a single logical test with no sub units.

Reply:
I think, TestRunMultipleContainersWithSameImage and TestRunMultipleContainersWithDifferentImages could be merged. Both of them are just calling runMultipleWebContainers.